src/envs/painting.py

```diff
@@ -85,65 +85,59 @@ def __init__(self) -> None:
-            initiable=self._handempty_initiable,
-            terminal=utils.onestep_terminal)
-        self._Wash = ParameterizedOption(
+            initiable=self._handempty_initiable)
+        self._Wash = utils.SingletonParameterizedOption(
             # variables: [robot]
             # params: []
             "Wash",
+            self._Wash_policy,
             types=[self._robot_type],
             params_space=Box(0, 1, (0, )),  # no parameters
-            policy=self._Wash_policy,
-            initiable=self._holding_initiable,
-            terminal=utils.onestep_terminal)
-        self._Dry = ParameterizedOption(
+            initiable=self._holding_initiable)
+        self._Dry = utils.SingletonParameterizedOption(
             # variables: [robot]
             # params: []
             "Dry",
+            self._Dry_policy,
             types=[self._robot_type],
             params_space=Box(0, 1, (0, )),  # no parameters
-            policy=self._Dry_policy,
-            initiable=self._holding_initiable,
-            terminal=utils.onestep_terminal)
-        self._Paint = ParameterizedOption(
+            initiable=self._holding_initiable)
+        self._Paint = utils.SingletonParameterizedOption(
             # variables: [robot]
             # params: [new color]
             "Paint",
+            self._Paint_policy,
             types=[self._robot_type],
             params_space=Box(-0.01, 1.01, (1, )),
-            policy=self._Paint_policy,
-            initiable=self._holding_initiable,
-            terminal=utils.onestep_terminal)
-        self._Place = ParameterizedOption(
+            initiable=self._holding_initiable)
+        self._Place = utils.SingletonParameterizedOption(
             # variables: [robot]
             # params: [absolute x, absolute y, absolute z]
             "Place",
+            self._Place_policy,
             types=[self._robot_type],
             params_space=Box(
                 np.array([self.obj_x - 1e-2, self.env_lb, self.obj_z - 1e-2],
                          dtype=np.float32),
                 np.array([self.obj_x + 1e-2, self.env_ub, self.obj_z + 1e-2],
                          dtype=np.float32)),
-            policy=self._Place_policy,
-            initiable=self._holding_initiable,
-            terminal=utils.onestep_terminal)
-        self._OpenLid = ParameterizedOption(
+            initiable=self._holding_initiable)
+        self._OpenLid = utils.SingletonParameterizedOption(
             # variables: [robot, lid]
             # params: []
             "OpenLid",
+            self._OpenLid_policy,
             types=[self._robot_type, self._lid_type],
             params_space=Box(0, 1, (0, )),  # no parameters
-            policy=self._OpenLid_policy,
-            initiable=self._handempty_initiable,
-            terminal=utils.onestep_terminal)
+            initiable=self._handempty_initiable)
         # Static objects (always exist no matter the settings).
```

Comment:
in this file (painting.py) can't we also wipe out the appearances of the `Always update the memory dict` stuff, like you did in playroom.py?

Reply:
also, wipe out appearances of `params_space=Box(0, 1, (0, )),  # no parameters` since it's default?